### PR TITLE
Publish website to gh-pages branch on pushes to master (#2950)

### PR DESCRIPTION
**Pull request by mwachs5**
### Contributor Checklist

- [ ] Did you add Scaladoc to every public function/method?
- [ ] Did you add at least one test demonstrating the PR?
- [ ] Did you delete any extraneous printlns/debugging code?
- [ ] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [ ] Did you state the API impact?
- [ ] Did you specify the code generation impact?
- [ ] Did you request a desired merge strategy?
- [ ] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

  - new feature/API          

#### API Impact

No impact

#### Backend Code Generation Impact

No IMpact

#### Desired Merge Strategy

  - Squash: The PR will be squashed and merged (choose this if you have no preference

#### Release Notes

Publish to GitHub Pages website on pushes to master

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (Bug fix: `3.4.x`, [small] API extension: `3.5.x`, API modification or big change: `3.6.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -65,8 +65,6 @@ jobs:
         uses: ./.github/workflows/install-circt
       - name: Check Formatting
         run: sbt scalafmtCheckAll
-      - name: Documentation
```
Comment (mwachs5) [inline .github/workflows/test.yml:68]: this was redundant with the website build doing it so I removed it... could put it back

File: website/README.md [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 This directory provides the meta-website for the [Chisel](https://github.com/freechipsproject/chisel3) project.
 
+
```
Comment (seldridge) [inline website/README.md:5]: Supreme nit: stray whitespace change.

File: website/build.sbt [hunk 3]
```diff
@@ -70,7 +70,7 @@ lazy val micrositeSettings = Seq(
   ghpagesNoJekyll := false,
   ghpagesRepository := file("build/gh-pages"),
   ghpagesBranch := "gh-pages",
-  git.remoteRepo := "git@github.com:freechipsproject/www.chisel-lang.org.git",
+  git.remoteRepo := "git@github.com:chipsalliance/chisel3.git",
```
Comment (seldridge) [inline website/build.sbt:73]: Nice. I didn't realize these were all pointing at the old organization. 👍

**Review by seldridge — APPROVED** [review]
Great!

Thanks for fighting to get this thing working.

File: website/README.md (hunk 2 above)
Comment (mwachs5) [inline website/README.md:5]:
```suggestion
```

File: website/README.md [hunk 4]
```diff
@@ -138,3 +138,13 @@ There are two targets for cleaning the build:
...
+# Website Deployment
+
+The Website is automatically deployed on every push to the `master` branch.
+The built website is pushed to the `gh-pages` branch, which is the source
+for what is hosted at https://chipsalliance.github.io/chisel3/.
+
+Note that currently there is another website, https://www.chisel-lang.org/,
+the source for which is https://github.com/freechipsproject/www.chisel-lang.org.
+Following the 3.6 release, the chisel-lang.org be changed to be sourced from this repository.
```
Comment (mwachs5) [inline website/README.md:150]:
```suggestion
Following the 3.6 release, the chisel-lang.org URL will be changed to be sourced from this repository.
```

**Pull request merged**